### ISSUE TITLE
the loss is 4~6

### ISSUE THREAD

**Issue by QueenJuliaZxx** [thtrieu/darkflow#462]
hi,I really want to know why my loss is just between 4 and 6,and wont go down anymore?

**Comment by tlsgb456** [comment]
Default learning rate is 1e-5. How about add parameter --lr lower than default learning rate? For example, --lr 1e-6.

**Comment by joaopalma5** [comment]
@QueenJuliaZxx  I have the same problem, you solve it? How? I'm training with adam and you? My dataset have ~200 Images/Class and have 3 Classes.

First, I made 20 epochs with 3 images/class for overfitting and now run the trainer with 600 Images. I started train from yolo.weights

**Comment by sheerun** [comment]
I have the same problem. Lowering learning rate doesn't change anything

**Comment by joaopalma5** [comment]
@sheerun You can try this: https://github.com/thtrieu/darkflow/issues/473
first overfit with a few images, after go train with a large dataset

don't forget, share your experience with us

**Comment by sheerun** [comment]
For now I've increased batch size to 32, as suggested by paper: https://arxiv.org/abs/1711.00489

The improvement is small, but visible (from 4.5-6.5) to (4.0-5.0)

**Comment by joaopalma5** [comment]
And the final results? you test with images out of the train? the loss is an indicator but not guarantee better results!

**Comment by sheerun** [comment]
I did following to get down to 3.8:

1. Initialize fresh model and train it with learning rate 0.01 and batch 5
2. After it's around 10 loss (quite fast), increase batch to 10 and learning rate to 0.0001
3. When it's around 4.5 increase batch to 40 and learning rate down too 0.00008

I guess it's all about increasing batch size and decreasing learning rate when moving average loss stops moving (e.g. if 2 minutes relative difference is less than 0.9). It certainly works for test images but's not perfect:

![](https://cdn.pbrd.co/images/H0tx7Ru.jpg)

![](https://cdn.pbrd.co/images/H0txGqq.jpg)

![](https://cdn.pbrd.co/images/H0ty5jv.jpg)
